### PR TITLE
Task/isc 10 create project models

### PR DESCRIPTION
**Pull request by jdrew-sans**

### CODE REVIEW

File: srv/isc-agent/plugins/tcp/http/models.py [hunk 1]
```diff
@@ -1,177 +1,95 @@
...
 
 mapper_registry = registry()
 
 Base = mapper_registry.generate_base()
 
 
-# Sigs model
-class Sigs(Base):
-    __tablename__ = 'sigs'
+class Signature(Base):
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/models.py:9]: Can we put this in alphabetical order

File: srv/isc-agent/plugins/tcp/http/models.py [hunk 2]
```diff
@@ -1,177 +1,95 @@
...
+        return self.pattern_string
 
-class HdrResponses(Base):
-    __tablename__ = 'hdr_responses'
+
+class HeaderResponse(Base):
+    __tablename__ = 'header_responses'
 
     id = Column(Integer, primary_key=True)
     sig_id = Column(Integer)
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/models.py:29]: Can we update this to be full name

File: srv/isc-agent/plugins/tcp/http/models.py [hunk 3]
```diff
@@ -1,177 +1,95 @@
...
     data_field = Column(Text)
 
     def __repr__(self):
-        return f'''HdrResponses(id={self.id!r}, SigID={self.SigID!r},
-        HeaderField={self.HeaderField}, dataField={self.dataField!r})'''
-
-
-class Paths(Base):
-    __tablename__ = 'paths'
+        return f"{self.__class__.__name__}({self.id}:{self})"
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/models.py:34]: May we add a space after the colon

File: srv/isc-agent/plugins/tcp/http/models.py [hunk 4]
```diff
@@ -1,177 +1,95 @@
...
+        return f"{self.__class__.__name__}({self.id}: {self})"
 
+    def __str__(self):
+        return self.pattern_string
 
-class HdrResponses(Base):
-    __tablename__ = 'hdr_responses'
+
+class HeaderResponse(Base):
+    __tablename__ = 'header_responses'
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/models.py:26]: Shouldnt be plural

File: srv/isc-agent/plugins/tcp/http/models.py [hunk 5]
```diff
@@ -1,177 +1,95 @@
...
 
     id = Column(Integer, primary_key=True)
     sig_id = Column(Integer)
     header_field = Column(Text)
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/models.py:30]: Let's remove the `_field` reference. We know its a field. Same for anywhere else using field.

File: srv/isc-agent/plugins/tcp/http/models.py [hunk 6]
```diff
@@ -1,177 +1,95 @@
...
-    cmd = Column(Text)
+    command = Column(Text)
     path = Column(Text)
-    useragent = Column(Text)
-    vers = Column(Text)
-    formkey = Column(Text)
-    formvalue = Column(Text)
+    user_agent = Column(Text)
+    version = Column(Text)
+    #had to change from metadata to data becuase metadata is reserved for the Metadata instance when using a declarative base class.
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/models.py:67]: Got it, you can delete this comment

File: srv/isc-agent/plugins/tcp/http/models.py [hunk 7]
```diff
@@ -1,177 +1,95 @@
...
-
-
-class Responses(Base):
-    __tablename__ = 'responses'
+class Response(Base):
+    __tablename__ = 'response'
 
     id = Column(Integer, primary_key=True)
-    rid = Column(Integer)
+    #Will need to be foreign key connecting to request_log_id
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/models.py:83]: Can remove comment

**Review by bmyrtil-sans — CHANGES_REQUESTED** [review]

File: srv/isc-agent/plugins/tcp/http/models.py (hunk 1 above)
Comment (jdrew-sans) [reply srv/isc-agent/plugins/tcp/http/models.py:9]: Yep

**Pull request merged**